src/app/auth/components/registerBox.js

```diff
@@ -0,0 +1,269 @@
+// @flow
+
+import React from 'react';
+
+export default class LoginBox extends React.Component {
+  state: {errorMessage: ?string}
+  email: HTMLInputElement
+  password: HTMLInputElement
+
+  constructor() {
+    super();
+    this.state = { errorMessage: null };
+  }
+
+  onSubmit(ev: Event) {
+    ev.preventDefault();
+    this.props.onSubmit({ email: this.email.value, password: this.password.value });
+  }
+
+  flashErrorMessage(message: string) {
+    this.setState({ errorMessage: message });
+  }
+
+  render(): React.Element<any> {
+    return <div className="login-box" id="login-box">
+      {/* TODO: Better flash message */}
+      { !this.state.errorMessage ? '' :
+        <div style={{ border: '2px solid #f00', background: '#fdd' }}>
+          {this.state.errorMessage}
+        </div> }
+      <div className="header">
+        <i className="fa fa-user-plus" />
+        {this.props.title}
+      </div>
+
+      <div className="body">
+      <form onSubmit={this.onSubmit.bind(this)}>
+        <div className="form-group has-feedback">
+          <label>First Name</label>
+          <input name="name_first" className="form-control" type="text" ref={ref => this.name_first = ref}/>
+          <i className="fa form-control-feedback"/>
+        </div>
+        <div className="form-group has-feedback">
+          <label>Last Name</label>
+          <input name="name_last" className="form-control" type="text" ref={ref => this.name_last = ref}/>
+          <i className="fa form-control-feedback"/>
+        </div>
+        <div className="form-group has-feedback">
+          <label>Affiliation</label>
+          <select className="form-control" name="affiliation_id" ref={ref => this.affiliation_id = ref}>
+            <option selected="selected" value="">=== Select affiliation ===</option>
+            <option value="1">CFC - CODE FOR CHARLOTTE</option>
```

Comment:
Hm.. which technically requires we get that list affiliation endpoint going.

What can be a strange issue: That list will be paginated but for the dropdown we need _all_ records to display. Should be simple to work around but it's something to keep in mind.